stock_batch_picking/models/stock_batch_picking.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+# © 2012-2014 Alexandre Fayolle, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from openerp import _, api, fields, models
+
+from openerp.exceptions import UserError
+
+
+class StockBatchPicking(models.Model):
+    """ This object allow to manage multiple stock.picking at the same time.
+    """
+    _name = 'stock.batch.picking'
+
+    name = fields.Char(
+        'Name',
+        required=True, index=True,
+        copy=False, unique=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self.env['ir.sequence'].next_by_code(
+            'stock.batch.picking'
+        ),
+    )
+
+    state = fields.Selection([
+        ('draft', 'Draft'),
+        ('assigned', 'Available'),
+        ('done', 'Done'),
+        ('cancel', 'Cancelled')],
+        string='State',
+        readonly=True, index=True, copy=False,
+        default='draft',
+        help='the state of the batch picking. '
+        'Workflow is draft -> assigned -> done or cancel'
+    )
+
+    date = fields.Date(
+        'Date',
+        required=True, readonly=True, index=True,
+        states={
+            'draft': [('readonly', False)],
+            'assigned': [('readonly', False)]
+        },
+        default=fields.Date.context_today,
+        help='date on which the batch picking is to be processed'
+    )
+
+    picker_id = fields.Many2one(
+        'res.users', 'Picker',
+        readonly=True, index=True,
+        states={
+            'draft': [('readonly', False)],
+            'assigned': [('readonly', False)]
+        },
+        help='the user to which the pickings are assigned'
+    )
+
+    picking_ids = fields.One2many(
+        'stock.picking', 'batch_picking_id', 'Pickings',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='List of picking managed by this batch.'
+    )
+
+    active_picking_ids = fields.One2many(
+        'stock.picking', 'batch_picking_id', 'Pickings',
+        readonly=True,
+        domain=[('state', 'not in', ('cancel', 'done'))],
+    )
+
+    notes = fields.Text('Notes', help='free form remarks')
+
+    move_ids = fields.One2many(
+        'stock.move',
+        readonly=True,
+        string='Related stock moves',
+        compute='_compute_move_ids'
+    )
+
+    pack_operation_ids = fields.One2many(
+        'stock.pack.operation',
+        readonly=True,
+        string='Related pack operations',
+        compute='_compute_pack_operation_ids'
+    )
+
+    pack_operation_product_ids = fields.One2many(
+        'stock.pack.operation',
+        string='Related pack operations',
+        compute='_compute_pack_operation_product_ids',
+        inverse=lambda *args, **kwargs: None,
```

Comment:
I should have added (and will add) a comment for that, without this inverse (but do nothing), the modifications on pack operation (in views, I put qty_done field editable) will not be saved.